### PR TITLE
Support cut event when validating presence of key

### PR DESCRIPTION
**Pull request by benjaminapetersen**
@jwforres @spadgett 

re this bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1374154

This ended up being easier than expected, the event handler just needed an extra if statement. `ng-required` wasn't a good fit because it would immediately invalidate the form on page load.  The `ng-keyup` route validates after user input.

### CODE REVIEW

File: dist/angular-key-value-editor.js [hunk 1]
```diff
@@ -258,15 +258,24 @@
...
+                    $scope.forms.keyValueEditor[uniqueForKey(unique, $index)].$setValidity('noKey', true);
+                    return;
+                  }
                   if(!!viewValue) {
                     $scope.forms.keyValueEditor[uniqueForKey(unique, $index)].$setValidity('noKey', true);
                   } else {
                     $scope.forms.keyValueEditor[uniqueForKey(unique, $index)].$setValidity('noKey', false);
                   }
+                },
+                cutEvent: function() {
```
Comment (jwforres) [inline dist/angular-key-value-editor.js]: looks like you need to rebuild the dist

Comment (benjaminapetersen) [inline dist/angular-key-value-editor.js]: ah, oop... though odd, i would have thought travis would catch that...

**Comment by benjaminapetersen** [comment]
rebuilt.

`ng-disabled` per we discussed:

```html
ng-value
ng-keyup="hasKey(entry, $index)"
ng-disabled="{{uniqueForValue(unique, $index)}}"
```

but this immediately makes the form invalid prior to user input.

`ng-cut` is a thing, I didn't know that:

```html
ng-value
ng-keyup="hasKey(entry, $index)"
ng-cut="hasKey(entry, $index)"
```

which works as well, except its actually unnecessary in this case.

**Comment by spadgett** [comment]
The example you pasted is a bit different than what we discussed (`ng-disabled` vs `ng-required`).

I'm not sure why `ng-required` would make the form invalid on load. There'd be no value, so key wouldn't be required, right?

**Comment by benjaminapetersen** [comment]
sorry, meant `ng-required` :)

**Comment by benjaminapetersen** [comment]
Oop, refresh on this one, been sitting a while.

File: dist/angular-key-value-editor.js (hunk 1 above)
Comment (benjaminapetersen) [inline dist/angular-key-value-editor.js]: dist is be clean. odd.

**Comment by benjaminapetersen** [comment]
Per this bug: https://bugzilla.redhat.com/show_bug.cgi?id=1374154

**Comment by benjaminapetersen** [comment]
Ping! I can't tell if the comments here require additional work.  Bug is fixed. Do we see a need to see `ng-required` as an option or does keeping the fix in JS work?

**Comment by benjaminapetersen** [comment]
Ping on this one, related to blocker bug https://bugzilla.redhat.com/show_bug.cgi?id=1374154

**Comment by spadgett** [comment]
I'd expect something like this

```
ng-required="!allowEmptyKeys && entry.value"
```

**Comment by benjaminapetersen** [comment]
Hmm. [TEST]?

**Comment by benjaminapetersen** [comment]
Updated to `ng-required` and key the error message off of `$error.required`, eliminated the `ng-keypress` and `hasKey()` functions.  Ran through the manual test files, I believe everything is still in order.

@spadgett if looks good, I'll merge & prep a PR to the web console.

**Pull request merged**